File: TransCoder_tokenized_test_set_functions/java/PROGRAM_PAGE_REPLACEMENT_ALGORITHMS_SET_2_FIFO.java
static int pageFaults ( int pages [ ], int n, int capacity ) {
  HashSet < Integer > s = new HashSet < > ( capacity );
  Queue < Integer > indexes = new LinkedList < > ( );
  int page_faults = 0;
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( s . size ( ) < capacity ) {
      if ( ! s . contains ( pages [ i ] ) ) {
        s . add ( pages [ i ] );
        page_faults ++;
        indexes . add ( pages [ i ] );
      }
    }
    else {
      if ( ! s . contains ( pages [ i ] ) ) {
        int val = indexes . peek ( );
        indexes . poll ( );
        s . remove ( val );
        s . add ( pages [ i ] );
        indexes . add ( pages [ i ] );
        page_faults ++;
      }
    }
  }
  return page_faults;
}

File: TransCoder_tokenized_test_set_functions/cpp/PROGRAM_PAGE_REPLACEMENT_ALGORITHMS_SET_2_FIFO.cpp
int pageFaults ( int pages [ ], int n, int capacity ) {
  unordered_set < int > s;
  queue < int > indexes;
  int page_faults = 0;
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( s . size ( ) < capacity ) {
      if ( s . find ( pages [ i ] ) == s . end ( ) ) {
        s . insert ( pages [ i ] );
        page_faults ++;
        indexes . push ( pages [ i ] );
      }
    }
    else {
      if ( s . find ( pages [ i ] ) == s . end ( ) ) {
        int val = indexes . front ( );
        indexes . pop ( );
        s . erase ( val );
        s . insert ( pages [ i ] );
        indexes . push ( pages [ i ] );
        page_faults ++;
      }
    }
  }
  return page_faults;
}